sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java

```diff
@@ -96,4 +132,245 @@ private SpecializedBlobClientBuilder prepareBuilder() {
 
         return builder;
     }
+
```

Comment:
Should the regular upload (the one that takes the data and length?) also be ported over to this class?

Reply:
think we are following:
```
In a broader spectrum the BlobClient is meant to contain all methods that aren't 1 to 1 with the REST specification where the specialized (Append, Block, and Page clients) are 1 to 1 with the REST specification.
```

and upload(data, length) is 1 to 1 to `put` blob with a length header?  in that case it is expected to be in specialized package